com.microsoft.mrtk.input/InteractionModes/InteractionModeManager.cs

```diff
@@ -32,9 +32,9 @@ public InteractionMode CurrentMode
             }
 
             [SerializeField]
-            private List<XRBaseControllerInteractor> interactors = new List<XRBaseControllerInteractor>();
+            private List<XRBaseInteractor> interactors = new List<XRBaseInteractor>();
```

Comment:
Does this need to be serialized? If not, we should be using `IXRInteractor`s instead of the base classes. (But if this does need to be serialized for whatever reason, I don't think we can serialize interfaces)

Reply:
I think it might mainly be used for the inspector view but I can do another pass over this.

Reply:
Took a stab at making the manager govern `IXRInteractors` this past weekend, but ultimately believe that doing so would drastically change the InteractionModeManagers behavior in an undesirable way.

To preface, the Interaction Mode Manager's job is to group interactors based on the 'controller' game object they belong to, and enable/disable the interactors as appropriate based on the 'Interaction Mode' that the controller is in. When we change the controller mapping to use `IXRInteractors`, we no longer have the guarantee that the interactor is a component. This can be worked around by directly register/unregistering the interactor from the `XRInteractionManager` in place of enabling/disabling the component, but this alters the Interaction Mode Manager's behavior from it's original intended purpose.

I think ultimately we should stick to using `XRBaseInteractor` for the time being to preserve the behavior of disabling gameobject components. In the future if there is a demand to build interactors which aren't based on top of `XRBaseInteractor`, we can revisit the best way to manage those components, but I do not want to deviate from the original intent of the `InteractionModeManager`

Reply:
Sounds fine to me.